src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorCompilationService.cs

```diff
@@ -53,16 +54,29 @@ public CompilationResult Compile(RelativeFileInfo file)
                 throw new ArgumentNullException(nameof(file));
             }
 
+            var filePath = file.FileInfo.PhysicalPath ?? file.RelativePath;
```

Comment:
We need both pieces of information, the full path for line pragmas and relative path for figuring out _ViewImports. The current API doesn't allow this.